### PR TITLE
Added a fix for NPE caused by empty response (for example network relate...

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
...d)

Together with @achudzik we've added a possible fix for the NPE caused by network issues.

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
We've been also working on adding some HTTP srever stubs so that you have some integration tests inside the app without the need to boot up ETCD. https://github.com/marcingrzejszczak/jetcd/tree/wireMockIntegration We arrived at a certain point where we ran out of time ;) If you are interested feel free to reuse.

File: src/test/java/jetcd/EtcdClientTest.java [hunk 1]
```diff
@@ -45,6 +48,18 @@ public void testGet() throws EtcdException {
       assertThat(e.getErrorCode()).isEqualTo(100);
     }
   }
+	
+  @Test
+  public void should_fail_when_etcd_server_is_down() throws EtcdException {
```
Comment (diwakergupta) [inline src/test/java/jetcd/EtcdClientTest.java]: please use naming conventions from existing tests, e.g. testFailureWithoutEtcdServer

**Comment by diwakergupta** [comment]
LG modulo minor comment. Haven't checked the build failure; might be unrelated. Thanks for the PR!

File: src/main/java/jetcd/EtcdClientImpl.java [hunk 2]
```diff
@@ -96,6 +96,9 @@ public void compareAndSwap(final String key, final String oldValue,
   private static final class EtcdErrorHandler implements ErrorHandler {
     @Override
     public Throwable handleError(final RetrofitError cause) {
+	    if (cause.getResponse() == null) {
```
Comment (diwakergupta) [inline src/main/java/jetcd/EtcdClientImpl.java]: turns out build failure is legit; checkstyle :-)

`EtcdClientImpl.java:99:1: File contains tab characters (this is the first instance).`

**Comment by marcingrzejszczak** [comment]
OMG How I hate checkstyle! Since you have only acceptance tests and I don't have a VBox or Docker image to install go and etcd I couldn't actually run the tests on my machine. You should add some integration tests that are working on stubs so people can actually run the tests.

And this checkstyle.... honestly at some point I wanted to just give up :P Anyway - now it passes

File: src/main/java/jetcd/EtcdClientImpl.java [hunk 3]
```diff
@@ -96,7 +96,10 @@ public void compareAndSwap(final String key, final String oldValue,
   private static final class EtcdErrorHandler implements ErrorHandler {
     @Override
     public Throwable handleError(final RetrofitError cause) {
-      return (EtcdException) cause.getBodyAs(EtcdException.class);
+        if (cause.getResponse() == null) {
```
Comment (diwakergupta) [inline src/main/java/jetcd/EtcdClientImpl.java:99]: We're using 2-space indents. Sorry for the nits!

**Comment by diwakergupta** [comment]
:+1: looking good, just one tiny indent issue! If it's too much trouble, don't worry and I can fix it later

**Comment by marcingrzejszczak** [comment]
Hi - will you ever merge it? :D

**Comment by diwakergupta** [comment]
Whoops sorry @marcingrzejszczak completely dropped the ball on this! Thanks for the patch!

**Pull request merged**